### PR TITLE
Add HowTo for creating and publishing image templates to OBS best practices guide

### PR DESCRIPTION
**Pull request by bgeuken**
Documentation parts that have already been written in the reference guide are
re-used via xincludes.

### CODE REVIEW

**Comment by bgeuken** [comment]
@adrianschroeter @hennevogel Please review

Note that the template page overview screenshot is just a placeholder for now. Since we don't have any templates on the OBS instance yet.
I will convert this to a blog entry later on But I thought it might be nice to have this in our docu as well.

File: xml/obs_best_practice_image_templates.xml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+ <figure>
+  <title>OBS templates page</title>
+  <mediaobject>
+   <imageobject>
+    <imagedata fileref="image_templates_template_page_placeholder.png" />
+   </imageobject>
+  </mediaobject>
+ </figure>
+ <section>
+  <title>Creating your own image templates</title>
```
Comment (mschnitzer) [inline xml/obs_best_practice_image_templates.xml]: should be `Creating Your Own Image Templates` (capitalization style)

File: xml/obs_best_practice_image_templates.xml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  <para>
+   Create a subproject of your home project.
+  </para>
+  <figure>
+   <title>Form for creating image template subproject</title>
```
Comment (mschnitzer) [inline xml/obs_best_practice_image_templates.xml]: should be `Form for Creating Image Template Subproject` (capitalization style)

File: xml/obs_best_practice_image_templates.xml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    Published image templates are fetched via a project's attribute. Any package
+    container living in a published project will be visible on the image
+    templates page.
+   </para>
+  </note>
+  <para>
+   Within that project create a new package. That will be your actual image template.
+  </para>
+  <figure>
+   <title>New image template</title>
```
Comment (mschnitzer) [inline xml/obs_best_practice_image_templates.xml]: should be `New Image Template` (capitalization style)

File: xml/obs_best_practice_image_templates.xml [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    <imageobject>
+     <imagedata fileref="image_templates_package.png" />
+    </imageobject>
+   </mediaobject>
+  </figure>
+  <para>
+   Add sources for your image configuration.
+  </para>
+  <figure>
+   <title>Overview of sources of custom image template</title>
```
Comment (mschnitzer) [inline xml/obs_best_practice_image_templates.xml]: * The two `of` here are... not sure this looks somehow strange. Can you maybe think a bit about a different naming?
* Please also use the capitalization style then: http://docserv.suse.de/documents/Styleguide/docu_styleguide/single-html/#sec.capitalization

File: xml/obs_best_practice_image_templates.xml [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+  <xi:include href="obs_image_templates.xml" xpointer="image_template_icon_paragraph" xmlns:xi="http://www.w3.org/2001/XInclude"/>
+  <para>
+   For a full list of image descriptions and general information about building
+   images with KIWI please refer to the <ulink url="http://opensuse.github.io/kiwi/">
+   KIWI project page</ulink> and the
+   <ulink url="https://doc.opensuse.org/projects/kiwi/doc/">KIWI cookbook</ulink>.
+  </para>
+ </section>
+ <section>
+  <title>Publishing image templates on the official image templates page</title>
```
Comment (mschnitzer) [inline xml/obs_best_practice_image_templates.xml]: Please use the capitalization style: http://docserv.suse.de/documents/Styleguide/docu_styleguide/single-html/#sec.capitalization

File: xml/obs_image_templates.xml [hunk 6]
```diff
@@ -8,7 +8,7 @@
 ]>
 <chapter id="cha.obs.image_templates">
  <title>Image templates</title>
```
Comment (mschnitzer) [inline xml/obs_image_templates.xml]: Should be `Image Templates`

File: xml/obs_best_practice_image_templates.xml (hunk 1 above)
Comment (bgeuken) [inline xml/obs_best_practice_image_templates.xml]: I can find both styles in our docu. But I am ok to change it.

File: xml/obs_best_practice_image_templates.xml (hunk 2 above)
Comment (bgeuken) [inline xml/obs_best_practice_image_templates.xml]: I think for figures I prefer the current format. And it seems like the whole text will be converted to uppercase letters anyway. Looking and our online docu

**Comment by mdeniz** [comment]
@mschnitzer did you re-review this?

File: xml/obs_best_practice_image_templates.xml [hunk 7]
```diff
@@ -53,6 +53,20 @@
    </mediaobject>
   </figure>
   <para>
+   Add the 'KIWI image build' repository to your project. This repository is
+   needed to build KIWI images in your project.
+   Go to the 'Repositories' tab, click on 'Add repositories' and chlick on the
+   'KIWI image build' checkbox.
+  </para>
+  <figure>
+   <title>Enabling the kIWI image build repository</title>
```
Comment (mschnitzer) [inline xml/obs_best_practice_image_templates.xml]: Please use the capitalization style.

File: xml/obs_best_practice_image_templates.xml (hunk 2 above)
Comment (mschnitzer) [inline xml/obs_best_practice_image_templates.xml]: @bgeuken please check: http://docserv.suse.de/documents/Styleguide/docu_styleguide/single-html/#sec.capitalization

We always use this style in our documentations.

Comment (bgeuken) [inline xml/obs_best_practice_image_templates.xml]: @mschnitzer Our documentation(s) can be found [here](http://openbuildservice.org/help/manuals/);-) And we use different styles right now for section and chapter titles right now.

And regarding the undertitles for figures, as mentioned before, they are converted to uppercase letters anyway.

Comment (mschnitzer) [inline xml/obs_best_practice_image_templates.xml]: Didn't recognize that this is a in a `figure` element.

**Pull request merged**